widget/embedded/src/components/WalletDerivationPathModal/WalletDerivationPathModal.tsx

```diff
@@ -0,0 +1,108 @@
+import type { PropTypes } from './WalletDerivationPathModal.types';
+
+import { i18n } from '@lingui/core';
+import { Button, Image, MessageBox, Select, TextField } from '@rango-dev/ui';
+import React, { useEffect, useState } from 'react';
+
+import { WIDGET_UI_ID } from '../../constants';
+import { getContainer } from '../../utils/common';
+import { WatermarkedModal } from '../common/WatermarkedModal';
+import {
+  LogoContainer,
+  Spinner,
+  WalletImageContainer,
+} from '../WalletModal/WalletModalContent.styles';
+
+import {
+  derivationPathInputStyles,
+  InputLabel,
+  InputsContainer,
+} from './WalletDerivationPathModal.styles';
+
+export function WalletDerivationPathModal(props: PropTypes) {
+  const { onClose, onConfirm, open, derivationPaths, image, type } = props;
+
+  const [selectedDerivationPathId, setSelectedDerivationPathId] = useState(
+    derivationPaths?.[0]?.id || ''
+  );
+  const [derivationPathIndex, setDerivationPathIndex] = useState('0');
+
+  const handleConfirm = () => {
+    const selectedDerivationPath = derivationPaths?.find(
+      (derivationPath) => derivationPath.id === selectedDerivationPathId
+    );
+    if (selectedDerivationPath) {
```

Comment:
Is there any case where `selectedDerivationPath` is `undefined`? If yes, shouldn't we throw an error or give user a ui feedback to show something went wrong?

Reply:
Considering that Confirm Button will be disabled if 'derivationPaths' or 'selectedDerivationPathId' or 'derivationPathIndex'  be 'undefined', 'selectedDerivationPath' should not be 'undefined' when calling this function. I improved that with type assertion.

Reply:
As we talked, that would be great to keep the runtime check and throw an error. bc the error should never happens, add a comment to mention it's an unreachable code.

Reply:
Fixed